js/load.js

```diff
@@ -130,98 +229,22 @@ function pkpCreateVueApp(createAppArgs) {
-	vueApp.component('PkpLineChart', LineChart);
-	vueApp.component('PkpComposer', Composer);
-	vueApp.component('PkpDateRange', DateRange);
-	vueApp.component('PkpFile', File);
-	vueApp.component('PkpFileAttacher', FileAttacher);
-	vueApp.component('PkpFileUploader', FileUploader);
-	vueApp.component('PkpFileUploadProgress', FileUploadProgress);
-	vueApp.component('PkpFilter', PkpFilter);
-	vueApp.component('PkpFilterAutosuggest', FilterAutosuggest);
-	vueApp.component('PkpFilterSlider', FilterSlider);
-	vueApp.component('PkpFilterSliderMultirange', FilterSliderMultirange);
-	vueApp.component('PkpList', List);
-	vueApp.component('PkpListItem', ListItem);
-	vueApp.component('PkpModal', Modal);
-	vueApp.component('PkpMultilingualProgress', MultilingualProgress);
-	vueApp.component('PkpOrderer', Orderer);
-	vueApp.component('PkpPagination', Pagination);
-	vueApp.component('PkpProgressBar', ProgressBar);
-	vueApp.component('PkpSearch', Search);
-	vueApp.component('PkpTable', Table);
-	vueApp.component('PkpTableCell', TableCell);
-	vueApp.component('PkpTooltip', Tooltip);
-
-	// Register Form components
-	vueApp.component('PkpForm', Form);
-	vueApp.component('PkpFieldArchivingPn', FieldArchivingPn);
-	vueApp.component('PkpFieldAutosuggestPreset', FieldAutosuggestPreset);
-	vueApp.component('PkpFieldBase', FieldBase);
-	vueApp.component('PkpFieldBaseAutosuggest', FieldBaseAutosuggest);
-	vueApp.component('PkpFieldColor', FieldColor);
-	vueApp.component('PkpFieldControlledVocab', FieldControlledVocab);
-	vueApp.component('PkpFieldHtml', FieldHtml);
-	vueApp.component('PkpFieldMetadataSetting', FieldMetadataSetting);
-	vueApp.component('PkpFieldOptions', FieldOptions);
-	vueApp.component('PkpFieldPreparedContent', FieldPreparedContent);
-	vueApp.component('PkpFieldPubId', FieldPubId);
-	vueApp.component('PkpFieldRadioInput', FieldRadioInput);
-	vueApp.component('PkpFieldRichText', FieldRichText);
-	vueApp.component('PkpFieldRichTextarea', FieldRichTextarea);
-	vueApp.component('PkpFieldSelect', FieldSelect);
-	vueApp.component('PkpFieldSelectIssue', FieldSelectIssue);
-	vueApp.component('PkpFieldSelectIssues', FieldSelectIssues);
-	vueApp.component('PkpFieldSelectSubmissions', FieldSelectSubmissions);
-	vueApp.component('PkpFieldSelectUsers', FieldSelectUsers);
-	vueApp.component('PkpFieldShowEnsuringLink', FieldShowEnsuringLink);
-	vueApp.component('PkpFieldText', FieldText);
-	vueApp.component('PkpFieldTextarea', FieldTextarea);
-	vueApp.component('PkpFieldUpload', FieldUpload);
-	vueApp.component('PkpFieldUploadImage', FieldUploadImage);
-
-	// Required by the URN plugin, to be migrated at some point to pkp prefix
-	vueApp.component('field-text', FieldText);
-	vueApp.component('field-pub-id', FieldPubId);
-
-	// Register ListPanel
-	vueApp.component('PkpListPanel', ListPanel);
+	// register all global components
+	const allGlobalComponents = VueRegistry.getAllComponents();
+	Object.keys(allGlobalComponents).forEach((componentName) => {
```

Comment:
I think you can replace by `Object.entries()`, then use `([name, component]) => ...`.